instrumentation/jaxrs/jaxrs-2.0/jaxrs-2.0-common/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jaxrs/v2_0/HandlerData.java

```diff
@@ -1,106 +1,42 @@
-import io.opentelemetry.context.Context;
-import io.opentelemetry.instrumentation.api.servlet.ServerSpanNaming;
-import io.opentelemetry.instrumentation.api.servlet.ServletContextPath;
-import io.opentelemetry.instrumentation.api.tracer.BaseTracer;
-import io.opentelemetry.instrumentation.api.tracer.ServerSpan;
 import io.opentelemetry.instrumentation.api.tracer.SpanNames;
 import io.opentelemetry.javaagent.bootstrap.jaxrs.ClassHierarchyIterable;
-import io.opentelemetry.javaagent.bootstrap.jaxrs.JaxrsContextPath;
-import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
 import java.lang.annotation.Annotation;
 import java.lang.reflect.Method;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
-import java.util.function.Supplier;
 import javax.ws.rs.HttpMethod;
 import javax.ws.rs.Path;
 
-public class JaxRsAnnotationsTracer extends BaseTracer {
-  public static final String ABORT_FILTER_CLASS =
-      "io.opentelemetry.javaagent.instrumentation.jaxrs2.filter.abort.class";
-  public static final String ABORT_HANDLED =
-      "io.opentelemetry.javaagent.instrumentation.jaxrs2.filter.abort.handled";
+public class HandlerData {
 
-  private static final JaxRsAnnotationsTracer TRACER = new JaxRsAnnotationsTracer();
-
-  public static JaxRsAnnotationsTracer tracer() {
-    return TRACER;
-  }
-
-  private final ClassValue<Map<Method, String>> spanNames =
+  private static final ClassValue<Map<Method, String>> serverSpanNames =
       new ClassValue<Map<Method, String>>() {
         @Override
         protected Map<Method, String> computeValue(Class<?> type) {
           return new ConcurrentHashMap<>();
         }
       };
 
-  public Context startSpan(Class<?> target, Method method) {
-    return startSpan(Context.current(), target, method);
-  }
-
-  public Context startSpan(Context parentContext, Class<?> target, Method method) {
-    // We create span and immediately update its name
-    // We do that in order to reuse logic inside updateSpanNames method, which is used externally as
-    // well.
-    SpanBuilder spanBuilder = spanBuilder(parentContext, "jax-rs.request", INTERNAL);
-    setCodeAttributes(spanBuilder, target, method);
-    Span span = spanBuilder.startSpan();
-    updateSpanNames(
-        parentContext, span, ServerSpan.fromContextOrNull(parentContext), target, method);
-    return parentContext.with(span);
-  }
+  private final Class<?> target;
+  private final Method method;
 
-  public void updateSpanNames(
-      Context context, Span span, Span serverSpan, Class<?> target, Method method) {
-    Supplier<String> spanNameSupplier = getPathSpanNameSupplier(context, target, method);
-    if (serverSpan == null) {
-      updateSpanName(span, spanNameSupplier.get());
-    } else {
-      ServerSpanNaming.updateServerSpanName(
-          context, ServerSpanNaming.Source.CONTROLLER, spanNameSupplier);
-      updateSpanName(span, SpanNames.fromMethod(target, method));
-    }
```

Comment:
I removed this special case of having a different name for the handler span name when there is no server span (now the handler span has the same name whether or not there is a server span, which is consistent with other instrumentations).